Services/DataX.Gateway/DataX.Gateway.Api/Controllers/GatewayController.cs

```diff
@@ -87,17 +88,29 @@ static GatewayController()
             var configPackage = FabricRuntime.GetActivationContext().GetConfigurationPackageObject("Config");
             var serviceEnvironmenConfig = configPackage.Settings.Sections["ServiceEnvironment"];
             var appInsightsIntrumentationKey = serviceEnvironmenConfig.Parameters["AppInsightsIntrumentationKey"].Value;
+            var testClientId = serviceEnvironmenConfig.Parameters["TestClientId"].Value;
             _IsUserInfoLoggingEnabled = IsUserInfoLoggingEnabled();
             _StaticLogger = new ApplicationInsightsLogger("GatewayILogger", new Microsoft.ApplicationInsights.TelemetryClient(new TelemetryConfiguration(KeyVault.GetSecretFromKeyvault(serviceKeyvaultName, appInsightsIntrumentationKey))), new ApplicationInsightsLoggerOptions());
+            try
+            {
+                _ClientWhitelist.Add(KeyVault.GetSecretFromKeyvault(serviceKeyvaultName, testClientId));

```

Comment:
let's make this test client id adding configurable so that for instances that do not want this, it can be disabled.

Reply:
That is what this code does. Even if this secret does not exist, the backend does not throw. Hence in that sense it is configurable.